src/Infrastructure/InternetDomainNameValidator.php

```diff
@@ -12,8 +12,17 @@
  */
 class InternetDomainNameValidator implements DomainNameValidator {
 
+	private const BANNED_DOMAINS = [
+		'example.com'
+	];
+
 	public function isValid( string $domain ): bool {
-		return checkdnsrr( $domain, 'MX' ) || checkdnsrr( $domain, 'A' ) || checkdnsrr( $domain, 'SOA' );
+		return !$this->isDomainBanned( $domain ) &&
+			( checkdnsrr( $domain, 'MX' ) || checkdnsrr( $domain, 'A' ) || checkdnsrr( $domain, 'SOA' ) );
+	}
+
+	private function isDomainBanned( string $domain ): bool {
+		return in_array( trim( $domain ), self::BANNED_DOMAINS );
```

Comment:
I don't understand why the `trim` is needed here, at this point of the application the domain name (coming from the email) should already be trimmed. But it doesn't hurt.

Maybe also add `strtolower` so people don't get away with `exAmple.cOM`